src/rpc/RpcChannel.cpp

```diff
@@ -51,6 +52,70 @@ using namespace google::protobuf::io;
 namespace Hdfs {
 namespace Internal {
 
+
+class SaslOutputWrapper {
+public:
+    SaslOutputWrapper(SaslClient * client, RpcClient *rpcClient) {
+        saslClient = client;
+        this->rpcClient = rpcClient;
+    }
+
+    std::string wrap(std::string data) {
+        if (!saslClient->isPrivate() && !saslClient->isIntegrity())
+            return data;
+
+        std::string rawdata = saslClient->encode(data.c_str(), data.length());
+        msg.set_state(RpcSaslProto_SaslState_WRAP);
+        msg.set_token(rawdata);
+        int totalLen;
```

Comment:
It is set below after data is read. No point to initialize it before then as it is not used.